apps/api-extractor/src/collector/Collector.ts

```diff
@@ -425,50 +424,54 @@ export class Collector {
       this._collectReferenceDirectives(astEntity);
     }
 
-    if (exportedName) {
-      entity.addExportName(exportedName);
+    if (exportName) {
+      if (parent) {
+        entity.addLocalExportName(exportName, parent);
+      } else {
+        entity.addExportName(exportName);
+      }
     }
 
     return entity;
   }
 
-  private _createEntityForIndirectReferences(
-    astEntity: AstEntity,
-    alreadySeenAstEntities: Set<AstEntity>
-  ): void {
-    if (alreadySeenAstEntities.has(astEntity)) {
-      return;
-    }
+  private _recursivelyCreateEntities(astEntity: AstEntity, alreadySeenAstEntities: Set<AstEntity>): void {
+    if (alreadySeenAstEntities.has(astEntity)) return;
     alreadySeenAstEntities.add(astEntity);
 
     if (astEntity instanceof AstSymbol) {
       astEntity.forEachDeclarationRecursive((astDeclaration: AstDeclaration) => {
         for (const referencedAstEntity of astDeclaration.referencedAstEntities) {
           if (referencedAstEntity instanceof AstSymbol) {
-            // We only create collector entities for root-level symbols.
-            // For example, if a symbols is nested inside a namespace, only the root-level namespace
-            // get a collector entity
+            // We only create collector entities for root-level symbols. For example, if a symbol is
+            // nested inside a namespace, only the namespace gets a collector entity. Note that this
+            // is not true for AstNamespaceImports below.
             if (referencedAstEntity.parentAstSymbol === undefined) {
-              this._createCollectorEntity(referencedAstEntity, undefined);
+              this._createCollectorEntity(referencedAstEntity);
             }
```

Comment:
Long-term, to fix bugs like https://github.com/microsoft/rushstack/issues/2808 as well as track whether items within namespaces are exported, we'll need to create collector entities for nested symbols. For example:

```
export declare namespace Foo {
  declare class Bar {}
  export declare class Baz extends Bar {}
}
```

Right now, `Foo`, `Bar`, and `Baz` will all be marked as exported, because there's only one collector entity for all three API items (the entity for `Foo`). In reality, only `Foo` and `Baz` should be marked as exported, and `Bar` and `Baz` should have their own collector entities.

I think this will need to be in a separate PR though, as it's a larger change.